From 706943376dab4fb50ee9c0a6a78a8ae771571d99 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 23 Jan 2018 18:49:46 +0000
Subject: [PATCH] Add support for a stateless configuration file

Allow udisks to look at /usr/share/defaults/udisks/udisks2.conf in
case the /etc/udisks/udisks2.conf file does not exist.

[pmccarty] Rebased for udisks 2.8.4
[kprabhuv] Rebased for udisks 2.9.1

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
Signed-off-by: Karthik Prabhu Vinod <karthik.prabhu.vinod@intel.com>
---
 src/udisksconfigmanager.c | 30 ++++++++++++++++++++++++------
 1 file changed, 24 insertions(+), 6 deletions(-)

diff --git a/src/udisksconfigmanager.c b/src/udisksconfigmanager.c
index 9558e27..81cd469 100644
--- a/src/udisksconfigmanager.c
+++ b/src/udisksconfigmanager.c
@@ -142,7 +142,9 @@ parse_config_file (UDisksConfigManager         *manager,
                    GList                      **out_modules)
 {
   GKeyFile *config_file;
-  gchar *conf_filename;
+  gchar *conf_filename = NULL;
+  gchar *user_conf_filename;
+  gchar *system_conf_filename;
   gchar *load_preference;
   gchar *encryption;
   gchar *module_i;
@@ -150,17 +152,32 @@ parse_config_file (UDisksConfigManager         *manager,
   gchar **modules_tmp;
 
   /* Get modules and means of loading */
-  conf_filename = g_build_filename (G_DIR_SEPARATOR_S,
+  user_conf_filename = g_build_filename (G_DIR_SEPARATOR_S,
                                     manager->config_dir,
                                     PACKAGE_NAME_UDISKS2 ".conf",
                                     NULL);
 
-  udisks_debug ("Loading configuration file: %s", conf_filename);
+  system_conf_filename = g_build_filename (G_DIR_SEPARATOR_S,
+                                    "/usr/share/defaults",
+                                    PACKAGE_NAME_UDISKS2 ".conf",
+                                    NULL);
+
+  udisks_debug ("Loading configuration file: %s", user_conf_filename);
 
   /* Load config */
   config_file = g_key_file_new ();
   g_key_file_set_list_separator (config_file, ',');
-  if (g_key_file_load_from_file (config_file, conf_filename, G_KEY_FILE_NONE, NULL))
+
+  if (g_key_file_load_from_file (config_file, user_conf_filename, G_KEY_FILE_NONE, NULL))
+  {
+        conf_filename = user_conf_filename;
+  }
+  else if (g_key_file_load_from_file (config_file, system_conf_filename, G_KEY_FILE_NONE, NULL))
+  {
+        conf_filename = system_conf_filename;
+  }
+
+  if (conf_filename)
     {
       if (out_modules != NULL)
         {
@@ -213,11 +230,12 @@ parse_config_file (UDisksConfigManager         *manager,
     }
   else
     {
-      udisks_warning ("Can't load configuration file %s", conf_filename);
+      udisks_warning ("Can't load configuration file %s", system_conf_filename);
     }
 
   g_key_file_free (config_file);
-  g_free (conf_filename);
+  g_free (user_conf_filename);
+  g_free (system_conf_filename);
 }
 
 static void
-- 
2.30.0

